src/gitsh.c

```diff
@@ -1,12 +1,25 @@
 #include <err.h>
+#include <ruby.h>
 #include <sysexits.h>
-#include <unistd.h>
 
 int
 main(int argc, char *argv[])
 {
-    if (execv(GITSH_RB_PATH, argv) == -1) {
-        err(EX_SOFTWARE, GITSH_RB_PATH);
+    int ruby_argc;
+    char *ruby_argv[];
+
+    ruby_argc = argc + 1;
+    ruby_argv = malloc((ruby_argc + 1) * sizeof(char *));
```

Comment:
Can we use `calloc(3)` instead of `malloc(3)` with multiplication? That way we don't have to worry about `ruby_argc` being a really, really large number and then underflowing.

Reply:
TIL, and thanks for the tip! The linked article was also a joy to read.

Reply:
Done!